sfs/fd/source.py

```diff
@@ -225,6 +225,71 @@ def point_dipole(omega, x0, n0, grid, *, c=None):
         _np.power(r, 2) * _np.exp(-1j * k * r)
 
 
+def point_circular_piston(omega, x0, n0, grid, R, *, c=None):
+    r"""Point source with circular piston far-field directivity.
+
+    Parameters
+    ----------
+    omega : float
+        Frequency of source.
+    x0 : (3,) array_like
+        Position of source.
+    n0 : (3,) array_like
+        Normal vector (direction) of the circular piston.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    R : float
+        Radius of circular piston.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    numpy.ndarray
+        Sound pressure at positions given by *grid*.
+
+    Notes
+    -----
+    .. math::
+
+        G(\x-\x_0,\w) = \frac{1}{2\pi}
+            \frac{J_1(\wc R \sin(\Theta))}{\wc R \sin(\Theta)}
+            \frac{\e{-\i\wc|\x-\x_0|}}{|\x-\x_0|}
+    with :math:`\Theta` more conveniently defined by its cosine
```

Comment:
sfs/fd/source.py:docstring of sfs.fd.source.point_circular_piston:20: WARNING: Explicit markup ends without a blank line; unexpected unindent.